benchmarks/src/main/kotlin/macrobenchmarks/ChannelProducerConsumerMonteCarloBenchmark.kt

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright 2016-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+@file:JvmName("ChannelProducerConsumerMonteCarloBenchmark")
+
+package macrobenchmarks
+
+import benchmarks.common.*
+import org.nield.kotlinstatistics.*
+import java.io.*
+import java.nio.file.*
+import kotlin.math.*
+import kotlin.random.*
+
+/**
+ * Total amount of threads in a benchmark
+ */
+private val THREADS = arrayOf(1, 2, 4, 8, 16, 18, 32, 36, 64, 72, 96, 108, 128, 144)
+/**
+ * Warm up iterations count
+ */
+private const val WARM_UP_ITERATIONS = 5
+/**
+ * Maximum number of calling [runIteration] function
+ */
+private const val MAX_ITERATIONS = 5_000
+/**
+ * The baseline for loop iterations each working thread does on average on each interaction with a channel
+ */
+private const val BASELINE_WORK = 50
+/**
+ * The max multiplier for the [BASELINE_WORK]
+ */
+private const val MAX_WORK_MULTIPLIER = 5.0
+/**
+ * Approximate total number of sent/received messages
+ */
+private const val APPROXIMATE_BATCH_SIZE = 100_000
+/**
+ * The threshold when a current benchmark configuration stops running
+ */
+private const val MONTECARLO_STOP_THRESHOLD = 0.01
+/**
+ * Number of iteration between checking if a current benchmark configuration should be stopped
+ */
+private const val ITERATIONS_BETWEEN_THRESHOLD_CHECK = 50
+/**
+ * Different types of creating workload in the benchmark: using balanced or unbalanced work load
+ */
+private val WITH_BALANCING = listOf(true, false)
+/**
+ * Should select be used in the benchmark or not
+ */
+private val WITH_SELECT = listOf(true, false)
+/**
+ * Benchmark output file
+ */
+private const val OUTPUT = "out/results_channel_producer_consumer_montecarlo.csv"
+/**
+ * Options for the new jvm instance
+ */
+private val jvmOptions = listOf<String>(/*"-Xmx64m", "-XX:+PrintGC"*/)
```

Comment:
Current default configuration is unacceptable. 
On 16 cores MBP the running time is estimated to be 9 hours. Nobody will ever run this benchmark is such a configuration. It should have a developer mode (from 1 to 10 minutes max running time) and "measurement mode" with 30-60 minutes max on a decent machine.